node/src/components/chain_synchronizer.rs

```diff
@@ -215,36 +245,37 @@ impl ChainSynchronizer {
 
     fn commit_upgrade(
         &self,
-        effect_builder: EffectBuilder<JoinerEvent>,
-        upgrade_block_header: BlockHeader,
+        effect_builder: EffectBuilder<REv>,
+        switch_block_header_before_upgrade: BlockHeader,
+        is_emergency_upgrade: bool,
     ) -> Effects<Event> {
-        info!("committing upgrade");
+        info!(%is_emergency_upgrade, "committing upgrade");
         let config = self.config.clone();
-        let cloned_upgrade_block_header = upgrade_block_header.clone();
+        let cloned_block_header = switch_block_header_before_upgrade.clone();
         async move {
             let chainspec_raw_bytes = effect_builder.get_chainspec_raw_bytes().await;
-            let upgrade_config = match config
-                .new_upgrade_config(&cloned_upgrade_block_header, chainspec_raw_bytes)
-            {
-                Ok(state_update) => state_update,
-                Err(error) => {
-                    error!(?error, "failed to get global state update from config");
-                    return Err(error.into());
-                }
-            };
+            let upgrade_config =
+                match config.new_upgrade_config(&cloned_block_header, chainspec_raw_bytes) {
+                    Ok(state_update) => state_update,
+                    Err(error) => {
+                        error!(?error, "failed to get global state update from config");
+                        return Err(error.into());
+                    }
+                };
             effect_builder
                 .upgrade_contract_runtime(upgrade_config)
                 .await
         }
-        .event(|result| Event::UpgradeResult {
-            upgrade_block_header,
+        .event(move |result| Event::UpgradeResult {
+            switch_block_header_before_upgrade,
+            is_emergency_upgrade,
```

Comment:
As discussed offline, I'd like to avoid changing the pattern in this way right now.  We can probably apply that style in a lot of places if we decide to move in that direction.